### ISSUE TITLE
Issue with tflite interpreter

### ISSUE THREAD

**Issue by vanajareedy0426** [tensorflow/tensorflow#52044]
Hi,
Training a model with MNIST database and converting to tflite, while inferencing I'm getting error.
My code details are as given below 
`
import tensorflow as tf
import tensorflow_datasets as tfds
(ds_train, ds_test), ds_info = tfds.load( 'mnist',
                                         split=['train', 'test'],
                                         shuffle_files=True,
                                         as_supervised=True,
                                         with_info=True,
                                         )
def normalize_img(image, label):
  return tf.cast(image, tf.float32) / 255., label

ds_train = ds_train.map(normalize_img, num_parallel_calls=tf.data.experimental.AUTOTUNE)
ds_train = ds_train.cache()
ds_train = ds_train.shuffle(ds_info.splits['train'].num_examples)
ds_train = ds_train.batch(128)
ds_train = ds_train.prefetch(tf.data.experimental.AUTOTUNE)
ds_test = ds_test.map(normalize_img, num_parallel_calls=tf.data.experimental.AUTOTUNE)

ds_test = ds_test.batch(128)
ds_test = ds_test.cache()
ds_test = ds_test.prefetch(tf.data.experimental.AUTOTUNE)

#model 
model = tf.keras.models.Sequential([
        tf.keras.layers.Flatten(input_shape=(28, 28)),
        tf.keras.layers.Reshape((1,784)),
        tf.keras.layers.Conv1D( 16, 1,use_bias=False,activation = 'relu' ),
		tf.keras.layers.Conv1D(16, 1,use_bias=False ,activation = 'relu' ),
		tf.keras.layers.Conv1D( 16, kernel_size=3,  strides=1, padding= "causal", dilation_rate=2**0, groups=16,use_bias=False,activation ='relu'),
		tf.keras.layers.Conv1D( 16, 1 ,use_bias=False) ,
		tf.keras.layers.Flatten(),
        tf.keras.layers.Dense(10)
        ])
model.compile(
    optimizer=tf.keras.optimizers.Adam(0.001),
    loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
    metrics=[tf.keras.metrics.SparseCategoricalAccuracy()],
)

model.fit(
    ds_train,
    epochs=1,
    validation_data=ds_test,
)
model.summary()
model.save("./model.h5")

converter = tf.lite.TFLiteConverter.from_keras_model(model)
tflite_model = converter.convert()

# Save the tflite model.
with open('model.tflite', 'wb') as f:
  f.write(tflite_model)
`

tflite interpreter code 
`import numpy as np
import tensorflow as tf

# Load the TFLite model and allocate tensors.
interpreter = tf.lite.Interpreter(model_path="./assignment.tflite")
interpreter.allocate_tensors()

# Get input and output tensors.
input_details = interpreter.get_input_details()
output_details = interpreter.get_output_details()
input_shape = input_details[0]['shape']
input_data = np.array(np.random.random_sample(input_shape), dtype=np.float32)
interpreter.set_tensor(input_details[0]['index'], input_data)
interpreter.invoke()
output_data = interpreter.get_tensor(output_details[0]['index'])
`

I'm Getting
Exception has occurred: RuntimeError

tensorflow/lite/kernels/conv.cc:349 input->dims->data[3] != filter->dims->data[3] (16 != 1)Node number 13 (CONV_2D) failed to prepare.

When I'm trying tflite convertion and tflite interpreter I'm getting error which is as mentioned .
I tried to debug the issue and understood that if I kept the output of dilated convolution  as the input for the next layers then I'm getting this error. (due to dilated convolution dependency )

1.How to resolve this error?
2. If we cannot do it directly with tflite , then is there any alternative method for post quantization of model such as given above code?

Any help will be greatly appreciable . Thanks in Advance!

**Comment by sushreebarsa** [comment]
@vanajareedy0426 
In order to expedite the trouble-shooting process here,Could you please fill the issue [template](https://github.com/tensorflow/tensorflow/issues/new/choose), please provide a code snippet to reproduce the issue reported here. Thanks!
